.github/actions/aurora-manage-cluster/action.yml

```diff
@@ -0,0 +1,202 @@
+    default: 'rds-ca-rsa2048-g1' # Default value from variable.tf
+
+  default-database-name:
+    description: 'The name for the automatically created database on cluster creation.'
+    default: 'camunda' # Default value from variable.tf
+
+  s3-backend-bucket:
+    description: 'Name of the S3 bucket to store Terraform state'
+    required: true
+  s3-bucket-region:
+    description: 'Region of the bucket containing the resources states'
+  tf-modules-revision:
+    description: 'Git revision of the tf modules to use'
+    default: 'main'
+    required: true
+
+  tf-modules-path:
+    description: 'Path where the tf Aurora modules will be cloned'
+    default: './.action-tf-modules/aurora/'
+    required: true
+
+  # inherited from https://github.com/hashicorp/setup-terraform/blob/main/action.yml
+  tf-cli-config-credentials-hostname:
+    description: 'The hostname of a HCP Terraform/Terraform Enterprise instance to place within the credentials block of the Terraform CLI configuration file. Defaults to `app.terraform.io`.'
+    default: 'app.terraform.io'
+    required: false
+  tf-cli-config-credentials-token:
+    description: 'The API token for a HCP Terraform/Terraform Enterprise instance to place within the credentials block of the Terraform CLI configuration file.'
+    required: false
+  tf-terraform-version:
+    description: 'The version of Terraform CLI to install. Defaults to `latest`.'
+    default: 'latest'
+    required: false
+  tf-terraform-wrapper:
+    description: 'Whether or not to install a wrapper to wrap subsequent calls of the `terraform` binary and expose its STDOUT, STDERR, and exit code as outputs named `stdout`, `stderr`, and `exitcode` respectively. Defaults to `true`.'
+    default: 'true'
+    required: false
+  awscli-version:
+    description: 'Version of the aws cli to use'
+    required: true
+    # renovate: datasource=github-releases depName=aws/aws-cli
+    default: "2.15.52"
+
+outputs:
+  aurora-endpoint:
+    description: 'The endpoint of the deployed Aurora cluster'
+    value: ${{ steps.apply.outputs.aurora_endpoint }}
+
+  terraform-state-url:
+    description: 'URL of the Terraform state file in the S3 bucket'
+    value: ${{ steps.utility.outputs.terraform-state-url }}
+
+  # Add all terraform outputs dynamically
+  all-terraform-outputs:
+    description: 'All outputs from Terraform'
+    value: ${{ steps.fetch_outputs.outputs.all_terraform_outputs }}
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Use Utility Actions
+      id: utility
+      # seehttps://github.com/orgs/community/discussions/41927 it's not possible to optimize this yet
+      # steps.uses  cannot access the github context.
+      # uses: ${{ github.action_repository }}/utility-action@${{ github.action_ref }}
+      # TODO: hard pin this one once it's merged
+      #uses: "camunda/camunda-tf-eks-module/utility-action@main"
+      uses: "./.github/actions/utility-action"
```

Comment:
is there a functional difference that speaks against using it's own ref vs main?

I don't directly see the issue, especially interesting for PRs as otherwise your changes will always be based on `main` for this particular part.

Reply:
but I mean your suggestion is to pin it to main, why not keep it as is?

Reply:
ah, because it would need a checkout to make it work. Understood.
In that case we can resolve this.